### PR TITLE
added hour worked function

### PR DESCRIPTION
**Pull request by Poketimi**
This pull request adds a new feature to the `CalendarViewModel` for calculating total worked hours per employee within a given time range, and introduces comprehensive unit tests to verify its correctness for various scenarios.

### Feature Addition

* Added a new method `calculateWorkedHours` to `CalendarViewModel` that computes the total worked hours for each employee based on event participation and presence, distinguishing between past and future events.

### Testing Improvements

* Added unit tests in `CalendarViewModelTest` to verify that `calculateWorkedHours` correctly calculates hours for:
  - Past events, respecting presence information.
  - Future events, assuming all participants are present.
  - Aggregation of hours from multiple events for a single user.
  
  Ai used
  
  #294

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt [hunk 1]
```diff
@@ -24,7 +24,8 @@ import kotlinx.coroutines.launch
 data class CalendarUIState(
     val events: List<Event> = emptyList(),
     val errorMsg: String? = null,
-    val isLoading: Boolean = false
+    val isLoading: Boolean = false,
+    val workedHours: List<Pair<String, Double>> = emptyList()
```
Comment (noa-flo) [inline app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: This could be a `Map<String, Double>` instead of a `List<Pair<String, Double>>`. Using a map makes lookups by user ID faster and the intent clearer. 
Since the `String` represents the user ID and the `Double` the number of worked hours, you should also add a comment explaining what the key and the value represent.

File: app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt [hunk 2]
```diff
@@ -124,4 +125,44 @@ class CalendarViewModel(
...
+
+          if (event.endDate.isBefore(now)) {
+            // Past event: check presence
+            event.presence.forEach { (userId, isPresent) ->
+              if (isPresent) {
+                workedHoursMap[userId] = workedHoursMap.getOrDefault(userId, 0.0) + durationHours
+              }
+            }
+          } else {
+            // Future event: assume participation
```
Comment (noa-flo) [inline app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: Maybe it could be good to be able to see distinctly both. So maybe return 2 lists, one with the already checked work time, and the other with the projected work time. Or add a bool param `compileResults` maybe ?
It could be useful for later.

Comment (Poketimi) [reply app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: I though on wether implementing that or not.

What I came up with was that everything needed to do the separation is accessible in the view too, to it can be split here.

If in the future, we need to reuse this function in a different context, it is more manageable as it is smaller and more straightforward.

Let me know what you think of it

File: app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt [hunk 3]
```diff
@@ -124,4 +125,44 @@ class CalendarViewModel(
...
+    viewModelScope.launch {
+      try {
+        val events =
+            eventRepository.getEventsBetweenDates(orgId = orgId, startDate = start, endDate = end)
+        val workedHoursMap = mutableMapOf<String, Double>()
+        val now = Instant.now()
+
+        events.forEach { event ->
+          val durationHours =
+              java.time.Duration.between(event.startDate, event.endDate).toMinutes() / 60.0
```
[the remark this answers is not in the document]
Comment (Poketimi) [reply app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: done in [2427eab](https://github.com/swent-agendapp/agendapp/pull/314/commits/2427eab1066b8544527c18b2905398bf196a1276)

File: app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt [hunk 4]
```diff
@@ -124,4 +125,44 @@ class CalendarViewModel(
...
   }
+
+  /**
+   * Calculates the total worked hours for each employee within a given time range.
+   *
+   * @param start The start of the time range.
+   * @param end The end of the time range.
+   */
+  fun calculateWorkedHours(start: Instant, end: Instant) {
+    val orgId = selectedOrganizationId.value ?: return
```
[the remark this answers is not in the document]
Comment (Poketimi) [reply app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: done in [a3351ad](https://github.com/swent-agendapp/agendapp/pull/314/commits/a3351ad8a1dd94fe5b0dc26c40cc700e4490b127)

[the remark this answers is not in the document]
Comment (noa-flo) [reply app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: I was thinking of a require statement but this is fine and even maybe better since the User is aware of the error witth this implementation !

File: app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt (hunk 2 above)
Comment (noa-flo) [reply app/src/main/java/com/android/sample/ui/calendar/CalendarViewModel.kt]: Well done ! But I don't think you added a test for the edge case where an event is currently happening to check if it is not counted 2 times in the past and in the future.

**Review by noa-flo — CHANGES_REQUESTED** [review]
There is just one comment still peding, the rest is all good ! Once it is resolved it should be good to go.

**Review by noa-flo — APPROVED** [review]
No more pending comment, I aprove !
